common/testutils/test_random.go

```diff
@@ -0,0 +1,108 @@
+package testutils
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// TestRandom provides all the functionality of math/rand.Rand, plus additional randomness functionality useful for testing
+// This struct wraps an instance of rand.Rand, and directly delegates all standard random methods to this internal instance
+type TestRandom struct {
+	internalRand *rand.Rand
```

Comment:
```suggestion
	*rand.Rand
```

Reply:
Use struct embedding instead! https://gobyexample.com/struct-embedding
This way you automatically "inherit" all the methods of rand.Rand, and don't need to write all those boilerplate proxy methods.

Reply:
Nice! Didn't know this was a thing

Reply:
[09e6390](https://github.com/Layr-Labs/eigenda/pull/976/commits/09e63904bb2ac75517d8e9ff4f0c25ea31cda94d)